### DISCUSSION TITLE
ACK not received in class C

### DISCUSSION THREAD

**Discussion by vAnArhist** [Lora-net/LoRaMac-node#1554] [Q&A]
Tested on HW: nrf52832+sx1262 module.
DIO1 used in interrupt mode, sx1262 used with TXCO, DIO3 as voltage source, DIO2 for TX/RX switch.

Tested stack version: v4.5.1, class C, US915 region
Network server: Comcast MachineQ server
LoraWAN gateway: Area8C
FW: no RTOS.

Facing issues with receiving ACK in confirmed messages mode.

Some words about our FW implementation:

1. Node connects to NS, get Join response, sends empty packets to increase data rate to DR>=2. Our packet size is 121byte, so it is the reason why we need to increase DR.
2. Node get UNIX timestamp from AS for internal use in FW.
3. Node TX 2 packets every 5minutes with 5sec pause between them.
4. All this time node samples data from ADC. Processing of LoRaMac events/radio interrupts execute in time when sampling is not done.

I have issue with receiving ACK from server for the second message, ACK for first message is ok.
I have IRQ_RX_TX_TIMEOUT for 2nd message generated before actual timeout for RX2 window (~3sec) passes.

Processing for ADC takes around ~520ms.
If i decrease this time to ~430ms ACK received, but not in all cases.

Questions:

1. Can anyone describe what is IRQ_RX_TX_TIMEOUT signal and why it can be generated? From the DS i see:

> When SymbNum is different from 0, the modem will wait for a total of SymbNum LoRa symbol to validate, or not, the correct detection of a LoRa packet. If the various states of the demodulator are not lock at this moment, the radio will generate the RxTimeout IRQ.

2. I dont understand difference between symbol timeout and lora header timeout. IRQ_RX_TX_TIMEOUT, as I understand, is symbol timeout, but RX cannot be performed if symbol timeout occur, so I dont understand how RX window will work in class C than.

**Comment by vAnArhist** [comment]
From SX1262 DS:
> In LoRa® mode, when going into Rx, the modem will lock as soon as a LoRa® symbol has been detected which may lead to
false detection. This phenomena is quite rare but nevertheless possible. To avoid this, the command
SetLoRaSymbNumTimeout can be used to define the number of symbols which will be used to validate the correct
reception of a packet.
When the SymbNum parameter is set the 0, the modem will validate the reception as soon as a LoRa® Symbol has been
detected.
When SymbNum is different from 0, the modem will wait for a total of SymbNum LoRa® symbol to validate, or not, the
correct detection of a LoRa® packet. If the various states of the demodulator are not lock at this moment, the radio will
generate the RxTimeout IRQ.

I see IRQ_RX_TX_TIMEOUT generation when sampling data performed right after TX. 
After decreasing time between packets from 5000ms (5s) to 500ms (0.5s) it started to work. Not sure how sampling can affect physical IRQ of GPIO pin of a chip, but without sampling logic this IRQ_RX_TX_TIMEOUT event not generating at all and each packet have ACK, even on join procedure.

Also, not sure why SetLoRaSymbNumTimeout set up for every receive window and how it`s calculated.

**Comment by Rgrndaal** [comment]
Bedankt voor uw email.
 
Ik ben afwezig tot maandag 28 augustus. E-mail gericht aan dit e-mailadres zal niet tussentijds worden gelezen of worden doorgestuurd. 
Voor dringende vragen verzoek ik u contact op te nemen met P. de Smit op ***@***.*** of 0162-511517.
Bedankt voor uw begrip. 

Met vriendelijke groet,
Remco Groenendaal
CLICKEY solutions
***@***.***

CLICKEY solutions: +31-162-511517
Lissenveld 47c, 4941 VL Raamsdonksveer, The Netherlands
www.clickey.eu

--------------------

Thank you for your e-mail. 

I am out of office until Monday August 28th. Your e-mail will not be read or forwarded until then. 
For urgent matters please contact P. de Smit at ***@***.*** or +31-162-511517.
Thank you for your understanding.

Kind regards,
Remco Groenendaal
CLICKEY solutions
***@***.***

CLICKEY solutions: +31-162-511517
Lissenveld 47c, 4941 VL Raamsdonksveer, The Netherlands
www.clickey.eu

**Comment by vAnArhist** [comment]
Any suggestions?
